app/gui2/src/composables/selection.ts

```diff
@@ -169,21 +135,58 @@ export function useSelection<T>(
     { predicate: (e) => e.target === e.currentTarget },
   )
 
-  return proxyRefs({
+  return {
+    // === Selected nodes ===
     selected,
-    anchor,
     selectAll: () => {
       for (const id of elementRects.keys()) selected.add(id)
     },
-    deselectAll: () => selected.clear(),
+    deselectAll: () => {
+      selected.clear()
+      deleted.clear()
+    },
     isSelected: (element: T) => selected.has(element),
-    isChanging,
     committedSelection,
     setSelection,
+    // === Selected nodes validity ===
+    elementDeleted: (id: T) => {
+      if (selected.delete(id)) deleted.add(id)
+    },
+    elementUndeleted: (id: T) => {
+      if (deleted.delete(id)) selected.add(id)
+    },
```

Comment:
We said during refinement, that we should have the `actually selected` field being a `computed` value which filters out `selected` which are not in `graphDb.nodeIdToNode.keys`. The keyset is reactive, so any change affecting our selection would make the computed to recompute.

I prefer the `computed` solution: it's maybe a bit farther from being optimal, but much simpler: wouldn't need to manually inform selection that a node was added/deleted.